examples/repo_example/succinct_hash_bin_delegations.py

```diff
@@ -0,0 +1,201 @@
+# smaller metadata files and thus a lower network overhead for repository-client
+# communication.
+#
+# The assignment of target files to a target's metadata is done automatically,
+# based on the byte digest of the target file name.
+#
+# The number of the bins, name prefix for all bins and key threshold are all
+# attributes that need to be configured.
+
+# Number of bins, bit length and bin number computation
+# -----------------------------------------------------
+# The right number of bins depends on the expected number of target files in a
+# repository. For the purpose of this example we choose:
+NUMBER_OF_BINS = 32
+#
+# The bit length is the number of bins that will be used to calculate the bin
+# for each target path. It can be calculated directly from NUMBER_OF_BINS:
+BIT_LENGTH = int(math.log2(NUMBER_OF_BINS))
+
+# Delegated role (bin) name format
+# --------------------------------
+# Each bin has a name in the format of f"{NAME_PREFIX}-{bin_number}".
+#
+# Name prefix is the common prefix of all delegated target roles (bins).
+# For our example it will be:
+NAME_PREFIX = "delegated_bin"
+#
+# The suffix "bin_number" is a zero-padded hexadecimal number of that
+# particular bin.
+
+# Keys and threshold
+# ------------------
+# Given that the primary concern of succinct hash bin delegation is to reduce
+# network overhead it was decided that all bins will be signed by the same
+# set of one or more signing keys.
+#
+# Before generating the keys a decision has to be made about the number of keys
+# required for signing the bins or in other words the value of the threshold.
+# For the purpose of this example we choose the threshold to be:
+THRESHOLD = 1
+# Note: If THRESHOLD is changed to more than 1 the example should be modified.
+
+
+def create_key() -> Tuple[Key, SSlibSigner]:
+    """Generates a new Key and Signer."""
+    sslib_key = generate_ed25519_key()
+    return Key.from_securesystemslib_key(sslib_key), SSlibSigner(sslib_key)
+
+
+key, signer = create_key()
+
+# Top level targets instance with succinct hash bin delegation
+# ------------------------------------------------------------
+# NOTE: See "Targets" and "Targets delegation" paragraphs in 'basic_repo.py'
+# example for more details about the Targets object.
+#
+# Now we have all the ingredients needed to create a Targets instance using
+# succinct hash bin delegation.
+#
+# First, we create a Targets metadata instance without any delegations.
+
+# We define expire as 7 days from today.
+expiration_date = datetime.utcnow().replace(microsecond=0) + timedelta(days=7)
+targets = Metadata(Targets(expires=expiration_date))
+
+# Then, we want to add delegations and with it information about the succinct
+# hash bin delegations which are represented by SuccinctRoles instance.
+#
+# Using succinct hash bin delegations has two restrictions:
+# 1) no other delegated roles have to be used
+# 2) only one succinct hash bin delegation can exist for one targets role
```

Comment:
I think you can omit this extra info. It is documented in TAP15 and in the metadata API. And not relevant for the example.